tests/EnumTest.php

```diff
@@ -303,8 +303,8 @@ private function assertJsonEqualsJson($json1, $json2)
     public function testSerialize()
     {
         // split string for Pretty CI: "Line exceeds 120 characters"
-        $bin = '4f3a33303a224d79434c6162735c54657374735c456e756d5c456e756d4669787'.
-            '4757265223a313a7b733a383a22002a0076616c7565223b733a333a22666f6f223b7d';
+        $bin = '4f3a33303a224d79434c6162735c54657374735c456e756d5c456e756d46697874757265223a323a7b733a38' .
+            '3a22002a0076616c7565223b733a333a22666f6f223b733a363a22002a006b6579223b733a333a22464f4f223b7d';
```

Comment:
Ha, so this test was added to avoid such BC breaks.

I'm afraid this could be a deal breaker. In short, already serialized enums may not be compatible with the new class. Is that right?

Reply:
Yes, was thinking about it as well and wanted to test it further to make sure.
I'm guessing the key will be null (https://3v4l.org/lILJ5) and a fallback will work to be implemented.

The second issue I'm not sure is that serialization will be bigger in terms of bytes. What do you think about that?

Reply:
The longer serialized string isn't a bc break to me, we never promised anything on the length of that string.

And right, if I understand correctly what you say we need a fallback for the null value.

Reply:
I think the test should have some comments then to make it more clear what it is for. It looked a lot like a [change detector test](https://testing.googleblog.com/2015/01/testing-on-toilet-change-detector-tests.html#:~:text=This%20is%20a%20change%2Ddetector,original%20or%20modified%20production%20code.).

Maybe a data provider `dpSerializedStringThatCanBeUnserialzed`.

Reply:
In case we have to deal with unserializing Enums that doesn't have a key, I moved the logic in `__wakeup()`.
I kept the unserializing test using the old format, adding equal comparison.
Let me know if that looks fine.